src/components/executive-education-2u/UserEnrollmentForm.jsx

```diff
@@ -85,16 +95,15 @@ const UserEnrollmentForm = ({
         ...(enableDataSharingConsent ? { dataShareConsent: !!values.dataSharingConsent } : {}),
       });
 
-      await sendEnterpriseTrackEventWithDelay(
-        enterpriseId,
-        'edx.ui.enterprise.learner_portal.executive_education.checkout_form.submitted',
-      );
-
-      setIsEnrollmentSubmitted(true);
-      onCheckoutSuccess(result);
+      await onSuccess(result);
     } catch (error) {
-      setFormSubmissionError(error);
-      logError(error);
+      const httpErrorStatus = error?.customAttributes?.httpErrorStatus;
+      if (httpErrorStatus === 422 && error?.message?.includes('User has already purchased the product.')) {
+        await onSuccess(error.message);
+      } else {
+        setFormSubmissionError(error);
+        logError(error);
```

Comment:
[nit/suggestion] it may be a good idea to also track the 422 status code case as well, perhaps with a `logInfo` containing a message that includes the enterprise UUID and LMS user id, so we have a bit more observability into when this happens. That said, this code path likely won't be exercised much after the mid-June migration date to the EMET system so I defer to you on whether adding the `logInfo` is worth it :)

Reply:
Good call, added!